packages/react-server/src/ReactFizzServer.js

```diff
@@ -891,6 +972,203 @@ function renderSuspenseBoundary(
   popComponentStackInDEV(task);
 }
 
+function replaySuspenseBoundary(
+  request: Request,
+  task: ReplayTask,
+  props: Object,
+  replayNode: ReplaySuspenseBoundary,
+): void {
+  pushBuiltInComponentStackInDEV(task, 'Suspense');
+
+  const previousReplaySet: ReplaySet = task.replay;
+
+  const parentBoundary = task.blockedBoundary;
+
+  const content: ReactNodeList = props.children;
+
+  const fallbackAbortSet: Set<Task> = new Set();
+  const resumedBoundary = createSuspenseBoundary(
+    request,
+    fallbackAbortSet,
+    task.keyPath,
+  );
+  resumedBoundary.parentFlushed = true;
+  // We restore the same id of this boundary as was used during prerender.
+  resumedBoundary.id = replayNode[4];
+  resumedBoundary.rootSegmentID = replayNode[5];
+
+  // We can reuse the current context and task to render the content immediately without
+  // context switching. We just need to temporarily switch which boundary and replay node
+  // we're writing to. If something suspends, it'll spawn new suspended task with that context.
+  task.blockedBoundary = resumedBoundary;
+  task.replay = {nodes: replayNode[3], pendingTasks: 1};
+  if (enableFloat) {
+    // Does this even matter for replaying?
+    setCurrentlyRenderingBoundaryResourcesTarget(
+      request.renderState,
+      resumedBoundary.resources,
+    );
+  }
```

Comment:
A child of this boundary may need to render and during that render new styles could be discovered so we'd want to associate those to this boundary. Makes sense to me that this is still needed